src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,149 @@
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void emptyLogin_NotOk() {
+        User user = createNewUser("", VALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerShortPassword_NotOk() {
+        User user = createNewUser(VALID_LOGIN, SHORT_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void passwordNullValue_NotOk() {
+        User user = createNewUser(VALID_LOGIN, null, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void emptyPassword_NotOk() {
+        User user = createNewUser(VALID_LOGIN, "", VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void ageNullValue_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void underAgeUser_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, UNDERAGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void negativeUserAge_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, NEGATIVE_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void zeroAgeValue_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, ZERO_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void largeAge_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, OVER_AGE);
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals(LARGE_AGE_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    void addNewUser_Ok() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        User actual = registrationService.register(user);
+        assertEquals(actual, user);
+        assertTrue(Storage.people.contains(user));
+    }
+
+    private User createNewUser(String login, String password, Integer age) {
+        User user = new User();
+        user.setLogin(login);
+        user.setPassword(password);
+        user.setAge(age);
+        return user;
+    }
```

Comment:
The helper method `createNewUser` is good for avoiding code duplication, but make sure it doesn't hide any potential issues with object creation that should be part of the tests, like creating a user with invalid data.